config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/bundle/builder/TrustedCertEntityBuilder.java

```diff
@@ -84,7 +86,8 @@
     }
 
     private Entity buildTrustedCertEntity(String name, TrustedCert trustedCert, Map<String, SupplierWithIO<InputStream>> certificateFiles, Document document) {
-        final String id = idGenerator.generate();
+        final String id = trustedCert.getAnnotatedEntity() != null && trustedCert.getAnnotatedEntity().getId() != null ?
+                trustedCert.getAnnotatedEntity().getId(): idGenerator.generate();
```

Comment:
Since this is repeatable, you can move this logic to generate Id in separate method.

Reply:
done